### PR TITLE
Feature/ Throw a warning if names are IN ALL CAPS

### PR DESCRIPTION
**Pull request by riturajj**
Added a new rule "AllCaps" which checks the name elements of Person, Candidate and Contest elements and throws out warnings if the names are in all upper case letters.

### CODE REVIEW

File: rules.py [hunk 1]
```diff
@@ -912,6 +912,47 @@ def check(self):
                 "The Election File has following issues with the identifiers.", error_log)
 
 
+class AllCaps(base.BaseRule):
+    """The Name elements in Candidates, Contests and Person elements should not be in all uppercase.
+
+    If the name elements in Candidates, Contests and Person elements are in uppercase, 
+    the list of bojectIds of those elements will be returned to the user as a warning."""
```
Comment (jdmgoogle) [inline rules.py]: bojectIds ~> objectIds

File: rules.py [hunk 2]
```diff
@@ -912,6 +912,47 @@ def check(self):
                 "The Election File has following issues with the identifiers.", error_log)
 
 
+class AllCaps(base.BaseRule):
+    """The Name elements in Candidates, Contests and Person elements should not be in all uppercase.
```
Comment (jdmgoogle) [inline rules.py]: Please use a sample XML file to make sure it catches the types of errors you expect.

File: rules.py [hunk 3]
```diff
@@ -912,6 +912,47 @@ def check(self):
...
+
+    If the name elements in Candidates, Contests and Person elements are in uppercase, 
+    the list of bojectIds of those elements will be returned to the user as a warning."""
+
+    def elements(self):
```
Comment (jdmgoogle) [inline rules.py]: The `elements()` function from some of the other checks may be better examples, actually.

E.g.,
https://github.com/google/election_results_xml_validator/blob/master/rules.py#L205
https://github.com/google/election_results_xml_validator/blob/master/rules.py#L223
https://github.com/google/election_results_xml_validator/blob/master/rules.py#L238
https://github.com/google/election_results_xml_validator/blob/master/rules.py#L481

You should be able to just specify a list with Candidate, Contest, and Person.

File: rules.py [hunk 4]
```diff
@@ -912,6 +912,47 @@ def check(self):
...
+                eligible_elements.append(element.get("name"))
+        return eligible_elements
+
+    def check(self, element):
+        object_id = element.get("objectId")
+        if element.tag == "Candidate":
+            ballot_name = element.find("BallotName")
+            if ballot_name.find("Text") is not None:
+                name = ballot_name.find("Text").text
+                if name is not None and name == name.upper():
```
Comment (jdmgoogle) [inline rules.py]: Does this check work for various UTF-8 text? E.g.,

NORMA VERÓNICA FLORES PABLO

**Comment by ccongchen** [comment]
Reverting this PR as the validator will throw all-caps warning when there's no Name element for one of the entities to be checked (Candidate, Party or Contest).

File: rules.py (hunk 4 above)
Comment (ccongchen) [reply rules.py]: Yes, it works with UTF-8.

**Pull request merged**